internal/format/base_visitor.go

```diff
@@ -0,0 +1,213 @@
+)
+
+type baseVisitor struct {
+	*printer
+
+	Failures []*text.Failure
+}
+
+func newBaseVisitor(indent string) *baseVisitor {
+	return &baseVisitor{printer: newPrinter(indent)}
+}
+
+//func (v *baseVisitor) VisitMessage(element *proto.Message)         {}
+//func (v *baseVisitor) VisitService(element *proto.Service)         {}
+//func (v *baseVisitor) VisitSyntax(element *proto.Syntax)           {}
+//func (v *baseVisitor) VisitPackage(element *proto.Package)         {}
+//func (v *baseVisitor) VisitOption(element *proto.Option)           {}
+//func (v *baseVisitor) VisitImport(element *proto.Import)           {}
+//func (v *baseVisitor) VisitNormalField(element *proto.NormalField) {}
+//func (v *baseVisitor) VisitEnumField(element *proto.EnumField)     {}
+//func (v *baseVisitor) VisitEnum(element *proto.Enum)               {}
+//func (v *baseVisitor) VisitComment(element *proto.Comment)         {}
+//func (v *baseVisitor) VisitOneof(element *proto.Oneof)             {}
+//func (v *baseVisitor) VisitOneofField(element *proto.OneOfField)   {}
+//func (v *baseVisitor) VisitReserved(element *proto.Reserved)       {}
+//func (v *baseVisitor) VisitRPC(element *proto.RPC)                 {}
+//func (v *baseVisitor) VisitMapField(element *proto.MapField)       {}
+//func (v *baseVisitor) VisitGroup(element *proto.Group)             {}
+//func (v *baseVisitor) VisitExtensions(element *proto.Extensions)   {}
+
+func (v *baseVisitor) AddFailure(position scanner.Position, format string, args ...interface{}) {
+	v.Failures = append(v.Failures, &text.Failure{
+		Line:    position.Line,
+		Column:  position.Column,
+		Message: fmt.Sprintf(format, args...),
+	})
+}
+
+func (v *baseVisitor) PWithInlineComment(inlineComment *proto.Comment, args ...interface{}) {
+	if inlineComment == nil || len(inlineComment.Lines) == 0 {
+		v.P(args...)
+		return
+	}
+	// https://github.com/emicklei/proto/commit/5a91db7561a4dedab311f36304fcf0512343a9b1
+	//if inlineComment.Cstyle {
+	//args = append(args, inlineComment.Lines[0])
+	//v.P(args...)
+	//for _, line := range inlineComment.Lines[1:] {
+	//v.P(line)
+	//}
+	//return
+	//}
+	args = append(args, ` //`, cleanCommentLine(inlineComment.Lines[0]))
+	v.P(args...)
+	for i, line := range inlineComment.Lines[1:] {
+		line = cleanCommentLine(line)
+		if line == "" && i != len(inlineComment.Lines)-1 {
+			v.P(`//`)
+		} else {
+			v.P(`//`, line)
+		}
+	}
+}
+
+func (v *baseVisitor) PComment(comment *proto.Comment) {
+	if comment == nil || len(comment.Lines) == 0 {
+		return
+	}
+	// https://github.com/emicklei/proto/commit/5a91db7561a4dedab311f36304fcf0512343a9b1
+	//if comment.Cstyle {
+	//for _, line := range comment.Lines {
+	//v.P(line)
+	//}
+	//return
+	//}
+	// this is weird for now
+	// we always want non-c-style after formatting
+	for i, line := range comment.Lines {
+		line = cleanCommentLine(line)
+		if line == "" && !(i == 0 || i == len(comment.Lines)-1) {
+			v.P(`//`)
+		} else {
+			v.P(`//`, line)
+		}
```

Comment:
Similar to above - we might be able to shorten this.

Reply:
See above, want to make sure we're right on this.

Reply:
Done